File: user-api/ts-sdk/src/sdk/models/operations/verifyemailwithtoken.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class VerifyEmailWithTokenQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=token" })
  token: string;
}


export class VerifyEmailWithTokenRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: VerifyEmailWithTokenQueryParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: shared.UserVerificationPayload;
}


export class VerifyEmailWithTokenResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: user-api/go-sdk/pkg/models/operations/verifyemailwithtoken.go
package operations

import (
	"openapi/pkg/models/shared"
)

type VerifyEmailWithTokenQueryParams struct {
	Token string `queryParam:"style=form,explode=true,name=token"`
}

type VerifyEmailWithTokenRequest struct {
	QueryParams VerifyEmailWithTokenQueryParams
	Request     *shared.UserVerificationPayload `request:"mediaType=application/json"`
}

type VerifyEmailWithTokenResponse struct {
	ContentType string
	StatusCode  int64
}
